src/Umbraco.Web/Models/MediaCropDataSet.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Dynamic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Web;
+using Newtonsoft.Json;
+using Umbraco.Core;
+using Umbraco.Core.Dynamics;
+using Umbraco.Core.Models;
+using Umbraco.Core.Serialization;
+using Umbraco.Web.PropertyEditors.ValueConverters;
+
+namespace Umbraco.Web.Models
+{
+    [JsonConverter(typeof(NoTypeConverterJsonConverter<MediaCropDataSet>))]
+    [TypeConverter(typeof(MediaCropDataSetConverter))]
+    [DataContract(Name="mediaCropDataSet")]
+    public class MediaCropDataSet : ImageCropDataSet, IHtmlString, IEquatable<MediaCropDataSet>
+    {
+        [DataMember(Name = "udi")]
+        public Udi Udi { get; set; }
+
+        public IPublishedContent MediaItem { get; set; }
+
+        public override string GetCropUrl(string alias, bool useCropDimensions = true, bool useFocalPoint = false, string cacheBusterValue = null)
+        {
+            return Src + base.GetCropUrl(alias, useCropDimensions, useFocalPoint, cacheBusterValue);
+        }
+
+        /// <summary>
+        /// Indicates whether the current object is equal to another object of the same type.
+        /// </summary>
+        /// <returns>
+        /// true if the current object is equal to the <paramref name="other"/> parameter; otherwise, false.
+        /// </returns>
+        /// <param name="other">An object to compare with this object.</param>
+        public bool Equals(MediaCropDataSet other)
+        {
+            if (ReferenceEquals(null, other)) return false;
+            if (ReferenceEquals(this, other)) return true;
+            return string.Equals(Udi, other.Udi) && Equals(FocalPoint, other.FocalPoint) 
```

Comment:
The `string.Equals(Udi, other.Udi)` comparison converts both values to strings: this should be changed to `Udi.Equals(other.Udi)` to skip this conversion from happening.
```suggestion
            return Udi.Equals(other.Udi) && Equals(FocalPoint, other.FocalPoint) 
```